cli/readInputs.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+// This function allows for increased functionality of the Vault helper tool
+// Reads one line of user input at a time
+func readInput() {
+	fmt.Print("# Enter command or press q to quit: ")
+	scanner := bufio.NewScanner(os.Stdin)
+	for scanner.Scan() {
+		result := scanner.Text()
+		newRes := strings.Split(result, " ")
+		command := newRes[0]
+		if (command == "write" || command == "w") && len(newRes) >= 3 {
```

Comment:
There's a lot of functional duplication between this block and the `app.Commands` specified in `main.go`. Could you think about whether there's a way to reuse the `Action`s specified there here? That way if you modify the handling of a command there (ex. separating out the validation stage as suggested) it will get modified for both the detached and interactive modes.
If you can't think of a way to easily do this, that's 100% OK. Don't spend any more time on it, instead could you document this as future work/technical debt in the README?

Reply:
I've documented this in a file under `docs/Further-Notes.md`. I'll try to work on it a little more in the meantime